src/pages/Profile.js

```diff
@@ -1,22 +1,58 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
+import { AuthContext } from '../context/AuthContext';
+import axios from 'axios';
 
 function Profile() {
-  return (
-    <>
-      <h1>Profielpagina</h1>
-      <section>
-        <h2>Gegevens</h2>
-        <p><strong>Gebruikersnaam:</strong> hardcoded-test</p>
-        <p><strong>Email:</strong> hardcoded@test.com</p>
-      </section>
-      <section>
-        <h2>Strikt geheime profiel-content</h2>
-        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab alias cum debitis dolor dolore fuga id molestias qui quo unde?</p>
-      </section>
-      <p>Terug naar de <Link to="/">Homepagina</Link></p>
-    </>
-  );
+    const [profileData, setProfileData] = useState({});
+    const { user } = useContext(AuthContext);
+
+    useEffect(() => {
+        const source = axios.CancelToken.source();
+
+        async function fetchProfileData() {
+            const token = localStorage.getItem('token');
+
+            try {
+                const result = await axios.get('http://localhost:3000/660/private-content', {
+                    headers: {
+                        "Content-Type": "application/json",
+                        Authorization: `Bearer ${token}`,
+                    },
+                    cancelToken: source.token,
+                });
+
+                setProfileData(result.data);
+            } catch (e) {
+                console.error(e);
+            }
+        }
+        void fetchProfileData();
+
+        return function cleanup() {
+            source.cancel();
+        }
+    }, []);
+
+    return (
+        <>
+            <h1>Profielpagina</h1>
+            <section>
+                <h2>Gegevens</h2>
+                <p><strong>Gebruikersnaam:</strong> {user.username}</p>
+                <p><strong>Email:</strong> {user.email}</p>
+            </section>
+
+            {Object.keys(profileData).length > 0 &&
```

Comment:
Hier kun je dan
```suggestion
            {profileData &&
```
van maken